src/consensus/encode.rs

```diff
@@ -622,34 +621,16 @@ impl Decodable for Vec<u8> {
         if len > MAX_VEC_SIZE {
             return Err(self::Error::OversizedVectorAllocation { requested: len, max: MAX_VEC_SIZE })
         }
-        let mut ret = Vec::with_capacity(len);
-        ret.resize(len, 0);
+        let mut ret = vec![0u8; len];
         d.read_slice(&mut ret)?;
         Ok(ret)
     }
 }
 
-impl Encodable for Box<[u8]> {
```

Comment:
Is this one even used anywhere?

Reply:
yes. `pub struct Script(Box<[u8]>);` so all operations on `script.0` are via the Box impl

Reply:
In fairness we might want to just do this impl directly on `Script`. I think I did it on `Box<[u8]>` so that all this mostly-boilerplate code would be in encode.rs (which is full of boilerplate code) rather than in script.rs (which is mostly actual functionality). But this isn't a strong argument.

Reply:
Already reverted this change back